airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDenormalizedDestination.java

```diff
@@ -67,6 +78,30 @@ protected Function<String, String> getTargetTableNameTransformer(final BigQueryS
     return namingResolver::getIdentifier;
   }
 
+  @Override
+  protected void putStreamIntoUploaderMap(AirbyteStream stream, UploaderConfig uploaderConfig,
+      Map<AirbyteStreamNameNamespacePair, AbstractBigQueryUploader<?>> uploaderMap) throws IOException {
+    Table existingTable = uploaderConfig.getBigQuery().getTable(uploaderConfig.getConfigStream().getStream().getNamespace(), uploaderConfig.getTargetTableName());
+    AbstractBigQueryUploader<?> uploader = BigQueryUploaderFactory.getUploader(uploaderConfig);
+    BigQueryRecordFormatter formatter = uploader.getRecordFormatter();
+
+    if (existingTable != null)
+    {
+      LOGGER.info("Target table already exists. Checking could we use the modern destination processing!");
+      com.google.cloud.bigquery.Schema existingTableSchema = existingTable.getDefinition().getSchema();
+      if (existingTableSchema != null && !existingTableSchema.equals(formatter.getBigQuerySchema())) {
```

Comment:
What worries me about this comparison is whether there are conditions under which `existingTableSchema.equals(formatter.getBigQuerySchema())` evaluates to false even though the target table has been created by the new formatter. I looked at the implementation of `equals` method in `com.google.cloud.bigquery.Schema` and from what I can see, the way it evaluates the comparison is by comparing the protobuf representation of fields:

```
public boolean equals(Object obj) {
    return obj == this || obj instanceof Schema && Objects.equals(toPb(), ((Schema) obj).toPb());
  }


  TableSchema toPb() {
    TableSchema tableSchemaPb = new TableSchema();
    if (fields != null) {
      tableSchemaPb.setFields(fields.toPb());
    }
    return tableSchemaPb;
  }
```

`toPb` method of a field looks like following:
```
  TableFieldSchema toPb() {
    TableFieldSchema fieldSchemaPb = new TableFieldSchema();
    fieldSchemaPb.setName(name);
    fieldSchemaPb.setType(type.name());
    if (mode != null) {
      fieldSchemaPb.setMode(mode);
    }
    if (description != null) {
      fieldSchemaPb.setDescription(description);
    }
    if (policyTags != null) {
      fieldSchemaPb.setPolicyTags(policyTags.toPb());
    }
    if (getSubFields() != null) {
      List<TableFieldSchema> fieldsPb = Lists.transform(getSubFields(), TO_PB_FUNCTION);
      fieldSchemaPb.setFields(fieldsPb);
    }
    return fieldSchemaPb;
  }
```

based on this code, `existingTableSchema.equals(formatter.getBigQuerySchema())` will evaluate to `false` under any of the following conditions:

1. user added a field to the destination table
2. user added a description to a field in the destination table
3. user changed `NULLABLE`, `REQUIRED`  or `REPEATED` mode of a field
4. user set policy tags
5. user renamed field
6. user changed field type

of the 6 conditions listed above, conditions 1 - 4 should not cause us to fallback to `ObsoleteArrayFormatter` or to fail.